setup.py

```diff
@@ -74,6 +74,11 @@ def find_sources():
     platform_specific['define_macros'] = [('WIN32', 1)]
     jni_md_platform = 'win32'
 
+elif sys.platform == 'cygwin':
```

Comment:
sys.platform on cygwin returns cygwin, but the compiler defines __WIN32__.  As most of the windows stuff uses __WIN32__. There has to be some modifications.